src/operator/mxnet_op.h

```diff
@@ -448,6 +449,47 @@ struct Kernel<OP, cpu> {
 #endif
   }
 
+  /*!
+   * \brief Launch a generic CPU kernel with native random generator.
+   * \tparam rnd CPU random generator
+   * \tparam Args Varargs type to eventually pass to the OP::Map() functoion
+   * \param N Number of iterations
+   * \param args Varargs to eventually pass to the OP::Map() functoion
+   */
+  template<typename GType, typename ...Args>
+  inline static void LaunchRNG(mshadow::Stream<cpu> *,
+                               common::random::RandGenerator<cpu, GType> *rnd,
+                               const int N, Args... args) {
+    using namespace mxnet::common::random;
+#ifdef _OPENMP
+    const int omp_threads = std::min(kCPURndStateNum,
+        engine::OpenMP::Get()->GetRecommendedOMPThreadCount());
+    if (omp_threads < 2) {
```

Comment:
I don't think you need special code for omp_threads < 2. The general loop below will work there as well and not create any overhead either.

Reply:
refactored

Reply:
This was intentional.
OpenMP disables some compiler optimizations, so an omp loop with 1 thread is slower than a simple loop without omp